pricegraph/src/orderbook/scalar.rs

```diff
@@ -169,3 +170,52 @@ fn assert_strictly_positive_and_finite(value: f64) -> f64 {
     debug_assert!(num::is_strictly_positive_and_finite(value));
     value
 }
+
+/// The exchange rate weight used by the pathfinding algorithm.
+#[derive(Clone, Copy, Debug, Default, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Weight(i128);
```

Comment:
A better high level description of this change is that we use fixed point arithmetic instead of floating point. Effectively we use a i128 to store 103 bits of the fractional part and 16 bits of the whole part.

It might be a good idea to use a real fixed point crate for this. I found https://crates.io/crates/fixed which looks good based on the documentation.

Reply:
The issue is we use specific "infinity" semantics required by `petgraph` but it's worth trying out.

Reply:
If you use the crate https://docs.rs/fixed/1.1.0/fixed/struct.FixedI128.html#method.from_num is probably relevant (there are a *lot* of traits on this type).

Reply:
The crate didn't help much since we only need to use `+`. I ended up just creating a type alias in order to make it very clear what is going on.

Reply:
What do you think of the `Fixed24x104 = i128` type alias to make it clear that its a fixed point number?

Reply:
That's fine. A comment would also be fine. It just took me some time when I first read the PR to realize it was doing fixed point arithmetic because that word wasn't mentioned anywhere.